apps/src/lib/node/ledger/tendermint_node.rs

```diff
@@ -454,16 +456,21 @@ async fn write_tm_genesis(
     genesis.genesis_time = genesis_time
         .try_into()
         .expect("Couldn't convert DateTimeUtc to Tendermint Time");
-    genesis.consensus_params.block = Some(block::Size {
-        // maximum size of a serialized Tendermint block
-        // cannot go over 100 MiB
-        max_bytes: (100 << 20) - 1, /* unsure if we are dealing with an open
-                                     * range, so it's better to subtract one,
-                                     * here */
-        // gas is metered app-side, so we disable it
-        // at the Tendermint level
-        max_gas: -1,
-    });
+    #[cfg(not(feature = "abcipp"))]
+    {
+        genesis.consensus_params.block = Some(block::Size {
+            // maximum size of a serialized Tendermint block
+            // cannot go over 100 MiB
+            max_bytes: (100 << 20) - 1, /* unsure if we are dealing with an
+                                         * open
+                                         * range, so it's better to subtract
+                                         * one,
+                                         * here */
+            // gas is metered app-side, so we disable it
+            // at the Tendermint level
+            max_gas: -1,
+        });
+    }
```

Comment:
hm why does this not work for `abcipp`? btw we should add some kind of `NOTE(feature = "abcipp")` here, in case people in the future try to nuke this config param from the code

Reply:
there is this separate block below, I just guarded this to get `make check-abcipp` etc to pass

```rust
    #[cfg(feature = "abcipp")]
    {
        genesis.consensus_params.timeout.commit =
            config.consensus_timeout_commit.into();
    }
```

Reply:
This was the error

```
error[E0308]: mismatched types
   --> apps/src/lib/node/ledger/tendermint_node.rs:457:38
    |
457 |       genesis.consensus_params.block = Some(block::Size {
    |  _____------------------------------___^
    | |     |
    | |     expected due to the type of this binding
458 | |         // maximum size of a serialized Tendermint block
459 | |         // cannot go over 100 MiB
460 | |         max_bytes: (100 << 20) - 1, /* unsure if we are dealing with an open
...   |
465 | |         max_gas: -1,
466 | |     });
    | |______^ expected struct `Size`, found enum `std::option::Option`
    |
    = note: expected struct `Size`
                 found enum `std::option::Option<Size>`
```

Reply:
I didn't look into it further than this, but I'd hope `genesis.consensus_params.block` can stay the same for ABCI++ (once ABCI++ is finished, and `tendermint-rs` updated for it fully)

Reply:
oh looks like it's just a matter of wrapping the value in an `Option`. you can add a separate let binding like so:
```rs
let size = block::Size {
    // maximum size of a serialized Tendermint block
    // cannot go over 100 MiB
    max_bytes: (100 << 20) - 1, /* unsure if we are dealing with an open
                                 * range, so it's better to subtract one,
                                 * here */
    // gas is metered app-side, so we disable it
    // at the Tendermint level
    max_gas: -1,
});
#[cfg(not(feature = "abcipp"))]
let size = Some(size);
genesis.consensus_params.block = size;
```

Reply:
that works!